crates/router/src/connector/noon/transformers.rs

```diff
@@ -330,17 +333,33 @@ impl TryFrom<&types::PaymentsAuthorizeRouterData> for NoonPaymentsRequest {
                 },
             });
 
-        let (subscription, tokenize_c_c) =
-            match item.request.setup_future_usage.is_some().then_some((
-                NoonSubscriptionData {
-                    subscription_type: NoonSubscriptionType::Unscheduled,
-                    name: name.clone(),
-                },
-                true,
-            )) {
-                Some((a, b)) => (Some(a), Some(b)),
-                None => (None, None),
-            };
+        let subscription = item
+            .request
+            .get_setup_mandate_details()
+            .map(|mandate_data| {
+                let max_amount = match &mandate_data.mandate_type {
+                    Some(data_models::mandates::MandateDataType::SingleUse(mandate))
+                    | Some(data_models::mandates::MandateDataType::MultiUse(Some(mandate))) => {
+                        conn_utils::to_currency_base_unit(mandate.amount, item.request.currency)
+                    }
+                    _ => Err(errors::ConnectorError::MissingRequiredField {
```

Comment:
Why do we need to throw error if mandate_data.mandate_type is None? We can simply return None

Reply:
When setup_mandate_details is present, our system will create a mandate id. But to create a mandate payment with noon`max_amount` is a required field, hence we are throwing an error here